synapse/storage/databases/main/event_push_actions.py

```diff
@@ -1340,6 +1349,53 @@ def remove_old_push_actions_that_have_rotated_txn(
             if done:
                 break
 
+    @wrap_as_background_process("_clear_old_push_actions_staging")
+    async def _clear_old_push_actions_staging(self) -> None:
+        """Clear out any old event push actions from the staging table for
+        events that we failed to persist.
+        """
+
+        # We delete anything more than an hour old, on the assumption that we'll
+        # never take more than an hour to persist an event.
+        delete_before_ts = self._clock.time_msec() - 60 * 60 * 1000
+
+        if self._started_ts > self._clock.time_msec() - delete_before_ts:
```

Comment:
I think both of these are absolute times, not offsets?

```suggestion
        if self._started_ts > delete_before_ts:
```

Reply:
oh yeah